applications/mne_sample_set_downloader/downloader.cpp

```diff
@@ -0,0 +1,216 @@
+*     * Neither the name of MNE-CPP authors nor the names of its contributors may be used
+*       to endorse or promote products derived from this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+* WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+* PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+* INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+* PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+* HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+* NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+*
+* @brief    Downloader class definition.
+*
+*/
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// INCLUDES
+//=============================================================================================================
+
+#include "Downloader.h"
+#include "Extract.h"
+#include "ui_downloader.h"
+
+//*************************************************************************************************************
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+#include <QFileDialog>
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+#ifdef _WIN32
+
+void Downloader::on_extractButton_clicked()
+{
+    QString zipPath = "\"" + QString(QDir::toNativeSeparators(ui->filePath->text())) + "\"";
+    readyToExtract(boolDownloadStatus, zipPath, m_qCurrentPath);
+    ui->extractButton->setEnabled(false);
+
+}
+
+void Downloader::checkZipperPath()
+{
+    ui->label->setText("7zip installed? Please check the path");
+    ui->extractButton->setVisible(true);
+    ui->extractButton->setEnabled(true);
+    ui->downloadButton->setVisible(false);
+    ui->downloadButton->setEnabled(false);
+    ui->progressBar->setVisible(false);
+    ui->filePath->setVisible(true);
+    ui->toolButton->setVisible(true);
+    ui->toolButton->setEnabled(true);
+}
+
+void Downloader::on_toolButton_clicked()
```

Comment:
please do not use underscores in function names. change to onToolButtonClicked. 

http://wiki.mne-cpp.org/index.php/Coding_Conventions_%26_Style